quipucords/api/source/tests_source.py

```diff
@@ -449,10 +449,13 @@ def test_retrieve(self):
         url = reverse('source-detail', args=(initial['id'],))
         response = self.client.get(url)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertIn('credentials', response.json())
-        creds = response.json()['credentials']
+        response_json = response.json()
+        self.assertIn('credentials', response_json)
+        creds = response_json['credentials']
 
         self.assertEqual(creds, [self.net_cred_for_response])
+        self.assertIn('hosts', response_json)
+        self.assertEqual(response_json['hosts'][0], '1.2.3.4')
```

Comment:
Would be could to have a test here where we try and create a source with other valid json not just an array of strings. Should have errors for that.

Reply:
Added a couple of tests for invalid data types of hosts array.